### PR TITLE
Publish builds done on dev branch

### PR DESCRIPTION
**Pull request by tiagomota**
[Sample](https://github.com/SwissBorg/akka-persistence-postgres/actions/runs/5609654446/jobs/10263654048)

### CODE REVIEW

File: .github/workflows/dev.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: Dev CI
+
+on:
+  push:
+    branches: [ dev ]
```
Comment (lomigmegard) [inline .github/workflows/dev.yaml:5]: question: is it different from `release.yml` ? Why couldn't we put the `dev` branch there?

Comment (tiagomota) [reply .github/workflows/dev.yaml:5]: To have better control 😅 I wanted this branch to only publish after running the tests. But yes, we can just add the `dev` branch to the `scala.yaml` and `release.yaml` and we should be good for what we want to do 😅

Comment (tiagomota) [reply .github/workflows/dev.yaml]: Simplified @ [1e2bf75](https://github.com/SwissBorg/akka-persistence-postgres/pull/352/commits/1e2bf75e43853c662cf4f80b5f2d68c0386e5bb9)

**Review by lomigmegard — APPROVED** [review]
Yeah it's a bit strange to have the release and the tests in separated flows. But not important for now as long as it does the job.

**Pull request merged**